File: llm-gateway/src/main/java/com/llm_gateway/controller/LlmController.java
package com.llm_gateway.controller;

import com.llm_gateway.dto.analyze.AnalyzeRequestDto;
import com.llm_gateway.dto.analyze.AnalyzeResponseDto;
import com.llm_gateway.dto.ask.AskRequestDto;
import com.llm_gateway.dto.ask.AskResponseDto;
import com.llm_gateway.service.LlmService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/llm")
@CrossOrigin(origins = "*")
public class LlmController {
    private final LlmService llmService;

    public LlmController(LlmService llmService){
        this.llmService = llmService;
    }

    @PostMapping("/analyze")
    public AnalyzeResponseDto analyzeComplexity(@RequestBody AnalyzeRequestDto request){
        return llmService.analyzeComplexity(request);
    }

    @PostMapping("/ask")
    public AskResponseDto generateResponse(@RequestBody AskRequestDto request){
        return llmService.generateResponse(request);
    }
}

File: backend/src/modules/llm/llm.controller.ts
import {
  Body,
  Controller,
  HttpCode,
  HttpStatus,
  Inject,
  Post,
  UseGuards,
} from '@nestjs/common';
import { ApiBearerAuth, ApiOkResponse } from '@nestjs/swagger';
import { JwtAuthGuard } from 'src/modules/auth/common/jwt-auth.guard';
import { AnalyzeRequestDto } from './dto/analyze-request.dto';
import { AskRequestDto } from './dto/ask-request.dto';
import { ModelResponseDto } from './dto/model-response.dto';
import { LlmServicePort } from './interface/llm.service.port';

@Controller('llm')
export class LlmController {
  constructor(
    @Inject(LlmServicePort)
    private readonly llmService: LlmServicePort,
  ) {}

  @HttpCode(HttpStatus.OK)
  @Post('/analyze')
  @ApiBearerAuth()
  @ApiOkResponse({ type: ModelResponseDto })
  @UseGuards(JwtAuthGuard)
  analyze(@Body() analyzeRequestDto: AnalyzeRequestDto) {
    return this.llmService.analyze(analyzeRequestDto);
  }

  @HttpCode(HttpStatus.OK)
  @Post('/ask')
  @ApiBearerAuth()
  @ApiOkResponse({ type: ModelResponseDto })
  @UseGuards(JwtAuthGuard)
  ask(@Body() askRequestDto: AskRequestDto) {
    return this.llmService.ask(askRequestDto);
  }
}
